### ISSUE TITLE
etcd problem

### ISSUE THREAD

**Issue by notxcain** [aweber/rabbitmq-autocluster#18]
Hi! 

I'm trying to use this plugin with `etcd`, but when I run `rabbitmq-plugins enable autocluster` I get this:
```
Applying plugin configuration to rabbit@s2003.local... failed.
Error: {badarg,[{erlang,list_to_integer,["tcp://172.17.0.3:2379"],[]},
                {autocluster_util,as_integer,1,
                                  [{file,"src/autocluster_util.erl"},
                                   {line,34}]},
                {autocluster_etcd,nodelist,0,
                                  [{file,"src/autocluster_etcd.erl"},
                                   {line,56}]},
                {autocluster,ensure_registered,2,
                             [{file,"src/autocluster.erl"},{line,97}]},
                {autocluster,init,0,[{file,"src/autocluster.erl"},{line,27}]},
                {rabbit,'-run_step/2-lc$^1/1-1-',1,[]},
                {rabbit,run_step,2,[]},
                {rabbit,'-run_boot_steps/1-lc$^0/1-0-',1,[]}]}
```
where `172.17.0.3:2379` is my `etcd` node.

The content of `rabbitmq.config` is
```
[{rabbit, [{loopback_users, []}]},{autocluster, [{backend, etcd}, {etcd_host, "etcd"}, {longname, true}]}].
```

What could be the problem?

**Comment by gmr** [comment]
The problem looks like the etcd port is defaulting to the tcp URL which is odd, I'm not quite sure how you'd get that without an ``ETCD_PORT`` environment variable of ``tcp://172.17.0.3:2379`` or a ``etcd_port`` config value of  ``tcp://172.17.0.3:2379``. The built-in defaults would not cause that to happen.

**Comment by gmr** [comment]
It's been quite some time since you opened the ticket, please let me know if you'd still like to try and troubleshoot this. I do not believe it's code related.

**Comment by gmr** [comment]
Ok, figure this out. Docker container linking is overriding the ``ETCD_PORT`` environment variable, making it the URI and setting it to:

ETCD_PORT='tcp://172.17.0.3:2379'

**Comment by gmr** [comment]
Addressed in 563acd3
